Feature: Adding Batch Thresholds to By Host Table

TheWitness opened the pull request:
This changes provides many updates and a very important feature:

* Dramatically speeds up the display of hosts pages.
* Moves a few elim related api's into grid_functions.php to support must faster queries of Batch Hosts and hosts under Host Load pages.  
* Properly sort elim fields, numeric, string and boolean.
* Some Cleanup in grid_functions.php for alignment and SQL formatting
* Some Cleanup for PSR spacing, line breaks, removing spaces at EOL, etc.
* Some SQL Injection protection by conversion to prepared statements.

Commit:
Initial RTM 10.2.0.12 based Cacti (1.2.15, plus several fixing), thold/syslog plugins
  With git ident attribute and copyright-year(2021)

ibm-open-source-bot committed:
Initial commit

Commit:
Publish RTM Server 10.2.0 FixPack 12 source code

ddb4github committed:
Publish RTM Server 10.2.0.12 branch source code

Commit:
Publish RTM Server develop branch(for 10.2.0.13) source code

kim-fitness committed:
platformcomputing/lsf-addon-L3-tracker#377

Commit:
Remove line-end space
Update syslog table schema to add partition definition
Update default email send-from addr

kim-fitness committed:
platformcomputing/lsf-addon-L3-tracker#393

kim-fitness committed:
platformcomputing/lsf-addon-L3-tracker#395

kim-fitness committed:
platformcomputing/lsf-addon-L3-tracker#398

kim-fitness committed:
platformcomputing/lsf-addon-L3-tracker#398

Commit:
Minor fixing to use hash array instead of multi-condition

Commit:
Merge few Cacti core security bugfixing

Commit:
Fix typo in grid_add_cluster.php

kim-fitness committed:
platformcomputing/lsf-addon-L3-tracker#395

kim-fitness committed:
Merge branch 'main' of github.com:IBM/ibm-spectrum-lsf-rtm-server

kim-fitness committed:
platformcomputing/lsf-addon-L3-tracker#395

kim-fitness committed:
Revert "platformcomputing/lsf-addon-L3-tracker#395"

This reverts commit 97817de4c7b66f1dfede4bf146b57ee5f91cd8f2.

kim-fitness committed:
platformcomputing/lsf-addon-L3-tracker#395

Commit:
Rename "Poller for LSF 10.1.0.12" to "Poller for LSF 10.1.0.13"
Remove three unused files
PHP error when heuristics-->cluster health-->export

Commit:
Fixed: correct db_fetch_cell to prepraed function for various columns

Commit:
Fixed: typo for 'Authenticaion'
Fixed: lost some tree node parent info during upgrade from pre-1.0
Fixed: rtm_plugin_upgrade_partition should always use forcever
Fixed: 10_2_0_12 partition upgrade use incorrect table name
Fixed: grid_add_cluster.php does not apply four parameters due to typo

Commit:
Update CHANGELOG sentences
Fixed: show empty if benchmark job exitcode is 0
Fixed: Show license service down message even non-server collected
  platformcomputing/lsf-addon-L3-tracker#408

Commit:
Remove duplicate indexes by percona advise
misc issues(index, auth, typo) for Pre-10.2 directly upgrade by CLI
audit_database.php miss to check text/mediumtext column due to typo
Remove duplicate items from CHANGELOG

Commit:
Update grid_functions.php
  platformcomputing/lsf-addon-L3-tracker#429

Commit:
Add pendjob collector status as a factor to justify LSF cluster status
Update license checkour/usage page to show service list

Commit:
Update grid_functions.php
  platformcomputing/lsf-addon-L3-tracker#442
  platformcomputing/lsf-addon-L3-tracker#443

Commit:
Update plugin version number for next fixpack

Commit:
Update grid_partitioning.php:
  The min_field of partitioned grid_jobs_finished is '0' if all job
  are killed from pending status.
  platformcomputing/lsf-addon-L3-tracker#410

Commit:
Update alert.format, correct '$Id $' to html comment forma
  platformcomputing/lsf-addon-L3-tracker#445

Commit:
Update user_domains.php for Cacti 4121
  platformcomputing/lsf-addon-L3-tracker#450

Commit:
Update grid_functions.php
  platformcomputing/lsf-addon-L3-tracker#451

Commit:
Fixed: resreq parser reject '=' if LSF_STRICT_RESREQ=Y
  platformcomputing/lsf-addon-L3-tracker#436

Commit:
Update grid_lsdashboard.php to support '+' in 'Service Domain' filter field
  platformcomputing/lsf-addon-L3-tracker#433

Commit:
New: grid_bzen.php support to display GPU Prohibited mode
Fixed "force index" issue since commit 0f46ad72
Update: gridalarms and gridblstat plugin version

Commit:
Fixed: jgroup/ugroup/project might be ignored to create graph if name end with '.0'
  platformcomputing/lsf-addon-L3-tracker#463
More fix grid_lsdashboard.php to support '+' in 'Service Domain' filter field
  platformcomputing/lsf-addon-L3-tracker#433

Commit:
Fix a general hook_api usage, return value for most 'api_plugin_hook_function' usage;
Correct few type for hook register;
Fix 'valid_host_fields' hook function
  platformcomputing/lsf-addon-L3-tracker#464

Commit:
Fixed: Miss to update LSF_CONFDIR during adding cluster for non-default LSF_STRICT_CHECKING settngs

Commit:
Fixed: job host/gpu rusage graph render slow or incorrect because PK is not complete in WHERE
  platformcomputing/lsf-addon-L3-tracker#483

Commit:
Support PHP 8.0/8.1 for RHEL 9.x:
Upgrade Cacti core to 1.2.23, thold to 1.6.1, syslog to 4.1:
Fixed: Update spectrum theme for new item in Cacti/thold/syslog
Fixed: Update RTM customized login/logout package
Fixed: Remove cactiTableTitle and cactiTableButton class from heuristics.css

Commit:
Fixed: remove extra '/' from some url

Commit:
Fixed: Performance: Ensure alerm->metrics query once during alarm-editing
  platformcomputing/lsf-addon-L3-tracker#481

Commit:
Fixed: increase grid_queues.maxJobs to 10
  platformcomputing/lsf-addon-L3-tracker#482

Commit:
Fixed: Skip which temp tables end with '_<timestamp>' during database struct backup
Fixed: Skip `'grid_heuristics_user_history_today_<timestamp>' table during database data backup
Fixed: add '-f' for most mysqldump exection to avoid backup failure if found bad/lost tables
  platformcomputing/lsf-addon-L3-tracker#494
Fixed: mysqldump/mysql output WARNING with '--port' option since MariaDB 10.6 on SLES 15.4 and Ubuntu 22.04
  platformcomputing/lsf-rtm-tracker#233
Fixed: <finished> table include some very old data if upgrade from pre-10.2
  platformcomputing/lsf-addon-L3-tracker#456
Fixed: RTM database backup-->mysqldump fail if log_bin enabled without extra privilege granted
  platformcomputing/lsf-rtm-tracker#110

Commit:
Update gridcstat.php for a PHP error

Commit:
Fixed: Empty log message when save expression alert with empty expression item
Fixed: make consistent for all cacti_log 'envion'

Commit:
Fixed: meta plugin dialog JS error
  platformcomputing/lsf-addon-L3-tracker#467

Commit:
Fixed: Add 'lic_services_feature_details' in daily maint optimize list
  platformcomputing/lsf-addon-L3-tracker#470

Commit:
Fixed: Update grid_functions.php for PHP error during add cluster and advocate stopped

Commit:
Fixed: Skip upgrade process for gridalarms fresh installation

Commit:
Update CopyRightYear to 2023 for FixPack 14

Commit:
Fixed: SQL Error if job run on multi-hosts and partition is disabled
  platformcomputing/lsf-addon-L3-tracker#511

Commit:
Fixed: correct a typo in license pluggable demo readme
  platformcomputing/lsf-addon-l3-tracker#500

Commit:
Fixed: SQL error for reserve word 'partition`
  platformcomputing/lsf-addon-L3-tracker#501

Commit:
Update the description of "Queue, Host Group and Application Purge Frequency" option:
  platformcomputing/lsf-addon-L3-tracker#514

Commit:
Support job export column 'dependCond'
  platformcomputing/lsf-addon-L3-tracker#515
Fixed: PHP8 error: strlen(): Passing null to parameter if partition is not enabled
  platformcomputing/lsf-rtm-tracker#226
Fixed: RTM backup fail due to typo in previous commit cea2b29d
  platformcomputing/lsf-rtm-tracker#110

Commit:
Fixed: extend 'resource_name' column length to 40
  platformcomputing/lsf-addon-L3-tracker#533

Commit:
Fixed: invalid xml file format

Commit:
Fixed: sync all user/group column length to 60
Fixed: sync all mailUser column length to 512
Fixed: update thold_data.name length to 150
Update audit_schema.sql before release
  platformcomputing/lsf-rtm-tracker#253

Commit:
New: Apply "Disable All Thresholds" setting to "Grid Alerts"
  platformcomputing/lsf-addon-L3-tracker#524
Fixed: correct graph_template_input_id data type in rtm.sql
  platformcomputing/lsf-rtm-tracker#253

Commit:
Fixed: PHP error if advocate service stopped
Fixed: incorrect height in Device->DataQuery->verbose output

Commit:
Merge selected fixing from upstream(1.2.23+)

Commit:
Fixed: Merge more fixing from upstream repo

Commit:
Merge Cacti 5390, 5304, 5136, 5137
Update gridalarms STATS log of disable to check alerts

Commit:
Merge Cacti 5438, Thold 604
  platformcomputing/lsf-addon-L3-tracker#563

Commit:
Merge plugin_thold 585
  platformcomputing/lsf-addon-L3-tracker#552

Conflicts:
	cacti/plugins/thold/CHANGELOG.md

ityuhui committed:
Add user/job priority and CWD in RTM job detail and job export

Signed-off-by: Hui Yu <ityuhui@gmail.com>

Commit:
Update version number for fixpack 15, grid/RTM/gridalarms plugins, init patch.filelst
  platformcomputing/lsf-rtm-tracker#262
Update grid_functions.php to avoid fairshare value is reset and re-assign
Fixed: Undefined variable if disable 'grid_jobhosts_collection'
  platformcomputing/lsf-addon-L3-tracker#443
Revert number_format_i18n to number_format in grid_get_rrdcfvalue function because  number_format_i18n does not support changing thousands_separator

Commit:
Merge Cacti 5468 to improvie LDAP DN usability
  platformcomputing/lsf-addon-l3-tracker#548

sbuser-git committed:
Update poller_options.php

The bug fixes include:
1.	PSR in multiple areas
2.	Correct issue where TIMEOUT values in options files not processed correctly
3.	Remove old options records from deleted services
4.	Better handle cases where no records are returned avoiding warnings in Cacti log
5.	Remove some legacy global variables $cnn_id
6.	Properly use set_config_option() for some stateful settings
7.	Avoid errors with regard to differing EXCLUDEALL option file parameters
8.	Better handling of options file processing.  Don’t process an options file unless the option is checked on the form

Signed-off-by: Stephanie Buser <quic_sbuser@quicinc.com>

Commit:
New: Support five columns from 'cpu/mem peak effi'
  platformcomputing/lsf-rtm-tracker/issues/216

ityuhui committed:
Add a tooltip for the column "Num GPUs" when the job runs on  MIG devices
  platformcomputing/lsf-rtm-tracker#252

Signed-off-by: Hui Yu <ityuhui@gmail.com>

Commit:
platformcomputing/lsf-addon-L3-tracker#590

Signed-off-by: ZHI YUAN LIU <liuzybj@cn.ibm.com>

Commit:
platformcomputing/lsf-addon-L3-tracker#589

Signed-off-by: ZHI YUAN LIU <liuzybj@cn.ibm.com>

Commit:
Fixed: Update grid_lsdashboard.php to support multi-lm-feature for one LS feature:
Change some db_fetch_xxxx function to db_fetch_xxxx_prepared
  platformcomputing/lsf-addon-L3-tracker#584
Fixed: PHP error: Use of undefined constant ls_count:
  platformcomputing/lsf-addon-L3-tracker#303
Update grid_functions.php to fix Cluster Dashboard tmp=mem issue
  platformcomputing/lsf-addon-L3-tracker#583

Commit:
Update grid_functions.php to enable table optimize once heuristics disabled
  platformcomputing/lsf-addon-L3-tracker#510

Commit:
Add a link "LSF on IBM Cloud" into top-right of main-console
  platformcomputing/lsf-addon-L3-tracker#597

Commit:
Fixed: update grid_add_trees.php to ensure case insensitive between cluster name and tree name
  platformcomputing/lsf-addon-L3-tracker#592

Commit:
Revert poller_gridblstat.php from 4e1f566f2 to rollback DELETE SQL
  platformcomputing/lsf-addon-l3-tracker#584

Commit:
Update poller_options.php to ensure clean up unused license option data
  platformcomputing/lsf-addon-L3-tracker#613

Commit:
Fixed: numJobs of PEND/RUNNING of Project, and avg mem/swap for RUNNING jobs
  platformcomputing/lsf-addon-l3-tracker#619

Commit:
platformcomputing/lsf-addon-L3-tracker#595

Signed-off-by: ZHI YUAN LIU <liuzybj@cn.ibm.com>

ityuhui committed:
Set the default data source profile when creating a new ELEM instance

platformcomputing/lsf-addon-L3-tracker#546

Signed-off-by: Hui Yu <ityuhui@gmail.com>

Commit:
Upgrade Cacti to 1.2.26, Thold to 1.8, Syslog to 4.1
  platformcomputing/lsf-rtm-tracker#280
Remove deprecated and unused aggregate_items.php, thold/thold_daemon.service, run, uuild_mo.sh and update-pot.sh
splice_rra.php, syslog->hi.mo
Few security and functional fixing from Cacti 1.2.x

ityuhui committed:
Add multi-tasks and exclusive mode to benchmark jobs

platformcomputing/lsf-rtm-tracker#285

Signed-off-by: Hui Yu <ityuhui@gmail.com>

ityuhui committed:
Fix the time format display for benchmark jobs

platformcomputing/lsf-rtm-tracker#290

Signed-off-by: Hui Yu <ityuhui@gmail.com>

Commit:
Update all url param ffeature value to support mult-feature
Show concat features in LM Feature column under Summary tab
Feature filter support both ' ' and ',' delimiter
Remove duplicate feature/token from auto-complete list
  platformcomputing/lsf-addon-L3-tracker#584

Commit:
Fixed: data_template_hash is reset after import thold template (Thold_PR_660)

Commit:
New: Show "View Breached Items" for thold that is created from customization thold template
New: Support more than 5 or more parameters for thold scripts
  platformcomputing/lsf-rtm-tracker#284

Commit:
Change LEFT/RIGHT related grid_blstat_users query from table to sub-query for performance
  platformcomputing/lsf-addon-l3-tracker#584

Commit:
Update poller_lichist.php to support to specify lichist table partno
  platformcomputing/lsf-addon-L3-tracker#653

Commit:
Backup RTM database struncture before data backup
  platformcomputing/lsf-addon-L3-tracker#649
Show duplicate job if a job run on multi exec host of the same hgroup
  platformcomputing/lsf-addon-L3-tracker#615

Commit:
Correct code comments, remove unused variable

Commit:
Merge Cacti 5143 to grid->grid_device_table_replace
  platformcomputing/lsf-rtm-tracker#280

Commit:
Move benchmark schema upgrade code from grid to benchmark self
Initial benchmark_version by schema
  platformcomputing/lsf-rtm-tracker#285

Commit:
Clean cluster related plugin data after remove cluster
  platformcomputing/lsf-rtm-tracker#254
Remove benchmark after remove cluster with delete graph/device
  platformcomputing/lsf-addon-L3-tracker#541
Rename gridalarms->logger() to rtm_logger() to avoid redeclare error

Commit:
Update changed plugin version by grid/upgrade/<ver>.php if plugin has not db changed
  platformcomputing/lsf-rtm-tracker#262

kim-fitness committed:
Update grid_functions.php

https://github.ibm.com/platformcomputing/lsf-addon-L3-tracker/issues/611

Commit:
'@' can not ignore mkdir func error under RHEL 9 + PHP 8.1
  platformcomputing/lsf-rtm-tracker#297

Commit:
Update grid_functions.php to resolve strtotime for PHP 8.x
  platformcomputing/lsf-addon-L3-tracker#666

Commit:
Update grid_functions.php to check strtotime return value
  platformcomputing/lsf-addon-L3-tracker#666

Commit:
Update heuristics_webapi.php to rollback number_format_i18n to number_format for calc
  platformcomputing/lsf-addon-L3-tracker#671

Commit:
Update heuristics/INFO to 10.2.0.15
  platformcomputing/lsf-addon-L3-tracker#671

ityuhui committed:
Rollback cacti csrf_secret.php path to fix the CSRF error after Cacti upgrade
  platformcomputing/lsf-rtm-tracker#303

Signed-off-by: Hui Yu <ityuhui@gmail.com>

Commit:
Use xxx_request_var to replace _REQUEST/_GET/_POST operation.
Use html_escape_request_var, html_escape and cacti_escapeshellarg for security report
  platformcomputing/lsf-rtm-tracker#300

Commit:
Update modified plugin version
  platformcomputing/lsf-rtm-tracker#300

Commit:
Update lichist plugin version by grid-->10_2_0_15.php
  platformcomputing/lsf-rtm-tracker#300

Commit:
Job GPU graphing fail for GPU 0 or partition disabled
  platformcomputing/lsf-rtm-tracker#304

Commit:
Typo when calc estimated time

Commit:
date filter is not work well in job->hostgraph
  platformcomputing/lsf-addon-L3-tracker#668

Commit:
Typo: Timespan is not switched to CUSTOM for first timeshift since PDB 153339

Commit:
Merge Cacti 5785
  platformcomputing/lsf-rtm-tracker#308

Commit:
Duplicate "Poller for LSF 9.1" due to typo
  platformcomputing/lsf-rtm-tracker#309

Commit:
SQL Error due to typo
  platformcomputing/lsf-rtm-tracker#300

Commit:
grid_add_trees.php log error if base_path is not configured in settings table
  platformcomputing/lsf-rtm-tracker#315

Commit:
SQL Error: Incorrect TIMESTAMP value in MySQL 8 for host/gpu rusage graphing
  platformcomputing/lsf-rtm-tracker#321

Commit:
SQL Error: Incorrect TIMESTAMP value in MySQL 8 for PEND->EXIT job during poller_lichist.php
  platformcomputing/lsf-rtm-tracker#321

Commit:
Fixed: Cannot redeclare ack_logging()
  platformcomputing/lsf-rtm-tracker#322

Commit:
Report error grid_queues_distrib doesn't exist when delete cluster
  platformcomputing/lsf-rtm-tracker#320

Commit:
Partial rollback RTC 235229, to avoid fairshare filter lost by whole main area refresh
  platformcomputing/lsf-rtm-tracker#326

Commit:
SQL Error: Column 'min_time' cannot be null in MySQL 8 for empty job sub-table partition
  platformcomputing/lsf-rtm-tracker#328

ityuhui committed:
Update the usage message for gridbenchmark

platformcomputing/lsf-rtm-tracker#285

Signed-off-by: Hui Yu <ityuhui@gmail.com>

Commit:
More fixing for platformcomputing/lsf-addon-L3-tracker#590

ityuhui committed:
Check if the variable is null befere passing it to strlen

lsf-rtm-tracker#310

Signed-off-by: Hui Yu <ityuhui@gmail.com>

ityuhui committed:
Use Base64 encoding for search filters to avoid character escapes

lsf-rtm-tracker#314

Signed-off-by: Hui Yu <ityuhui@gmail.com>

Commit:
strtotime(): Passing null to parameter #1 ($datetime) of type string when some disku poller is down
  platformcomputing/lsf-rtm-tracker#291

Signed-off-by: Jing Chen <jingchen@cn.ibm.com>

Commit:
replace strlen to rtm_strlen for PHP 8.x. shell_exec always return NULL if command fail
  platformcomputing/lsf-rtm-tracker#291

Signed-off-by: Jing Chen <jingchen@cn.ibm.com>

Commit:
lic_get_quorum_status return NULL when lic_services status is 3 without lic_servers records
  platformcomputing/lsf-rtm-tracker#291

Signed-off-by: Jing Chen <jingchen@cn.ibm.com>

Commit:
Update RTM version to 10.2.0.15, CopyRightYear to 2024 for PHP CLI and Page CopyRight area:
   platformcomputing/lsf-rtm-tracker#262

Signed-off-by: Jing Chen <jingchen@cn.ibm.com>

Commit:
Update CopyRightYear to 2024 for license header of changed files
Insert git attr ident for new files
  platformcomputing/lsf-rtm-tracker#262

Signed-off-by: Jing Chen <jingchen@cn.ibm.com>

Commit:
Update Cacti Core to 1.2.27 with few security fixing from 1.2.28
  platformcomputing/lsf-rtm-tracker#280

Signed-off-by: Jing Chen <jingchen@cn.ibm.com>

Commit:
Update DB schema to Cacti 1.2.27 for fresh installation
  platformcomputing/lsf-rtm-tracker#280
Update RTM version and CopyRightYear
  platformcomputing/lsf-rtm-tracker#262

Signed-off-by: Jing Chen <jingchen@cn.ibm.com>

Commit:
Insert git attr ident for missed file
  platformcomputing/lsf-rtm-tracker#262

Signed-off-by: Jing Chen <jingchen@cn.ibm.com>

Commit:
Correct grid_clusters.lsf_version default to 10

Signed-off-by: Jing Chen <jingchen@cn.ibm.com>

SMark-Black committed:
Update README.md

SMark-Black committed:
Update INFO links and sync other fixes

SMark-Black committed:
Merge pull request #44 from SMark-Black/main

Update INFO links and sync other fixes

SMark-Black committed:
Update IBMDCO.md

SMark-Black committed:
Sync multiple fixes

SMark-Black committed:
Merge pull request #45 from SMark-Black/main

Sync multiple fixes

SMark-Black committed:
Add Alma support

SMark-Black committed:
Merge pull request #46 from SMark-Black/main

Add AlmaLinux support

cwei-ca committed:
Fix hgroup dropdown filter with special char

Fix hgroup dropdown filter with special char & and | in 6 pages.

cwei-ca committed:
Merge several changes

1. replace deprecated FILTER_SANITIZE_STRING with FILTER_DEFAULT
2. Fix deprecated error on gridalarms trim()
3. Fix thold LSF event triggering not show saved values
4. Move lichist partition logic before process finished job data
5. revert PR #199 Use Base64 encoding for search filters to avoid character escapes

SMark-Black committed:
Sync grid_functions.php to test build issue

SMark-Black committed:
Copyright update, and 360, 228, 232

SMark-Black committed:
Merge branch 'IBM:main' into main

SMark-Black committed:
Merge pull request #48 from SMark-Black/main

Update copyright and include fixes for 360, 228, 232

SMark-Black committed:
Inconsistent Mem Peak Efficiency display 270

SMark-Black committed:
Merge branch 'IBM:main' into main

SMark-Black committed:
Merge pull request #49 from SMark-Black/main

Fix Inconsistent Mem Peak Efficiency display 270

cwei-ca committed:
Use bkill instead of forcekill for pend job over limit

TheWitness committed:
QA: Issues when the graph can not be found

Signed-off-by: TheWitness <thewitness@cacti.net>

TheWitness committed:
Feature: Creating new feature showing batch tholds

Signed-off-by: TheWitness <thewitness@cacti.net>

TheWitness commented:
@SMark-Black, there are some SQL issues on closure events.  I'll upload an addendum after I get back from vacation.

TheWitness committed:
QA: Fixing problem with Closure Events

Signed-off-by: TheWitness <thewitness@cacti.net>

TheWitness commented:
Okay, @SMark-Black, fixes are in and this is now fully tested.

SMark-Black commented:
@TheWitness cool, thanks.  we'll have a look

TheWitness commented:
@SMark-Black I'll have several more coming in the weeks and months to come.  Off hours stuff.

cwei-ca commented:
Plz remove the change in html.php that was already in PR [50](https://github.com/IBM/ibm-spectrum-lsf-rtm-server/pull/50).

cwei-ca commented on cacti/plugins/grid/grid_bhosts.php:
```diff
@@ -965,7 +977,7 @@ function applyFilter() {
 
 		if ($('#resource_str').length) {
-			strURL += '&resource_str=' + encodeURIComponent($('#resource_str').val());
+			strURL += '&resource_str=' + escape($('#resource_str').val());
```
`escape()` is deprecated 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/escape
That's why replaced by `encodeURIComponent`

TheWitness replied:
Thanks.  I'll update this.

TheWitness replied:
Fixed now.

cwei-ca commented on cacti/plugins/grid/grid_bhosts.php:
```diff
@@ -627,64 +654,80 @@ function grid_view_get_bhosts_records(&$sql_where, $apply_limits = true, $rows =
 	//cacti_log("DEBUG: SQL: $sql_query");
-	return db_fetch_assoc_prepared($sql_query, $sql_params);
+
+	return db_fetch_assoc($sql_query);
```
After install fix, got error in cacti.log
![Screenshot 2025-05-13 at 3 23 48 PM](https://github.com/user-attachments/assets/121c935b-536e-48ba-aa94-c3b3a4524726)
That lead to this line.  wonder should use `db_fetch_assoc_prepared`

TheWitness replied:
This is fixed now.

cwei-ca commented on cacti/plugins/grid/grid_lsload.php:
```diff
@@ -87,25 +56,25 @@ function grid_view_get_load_records(&$sql_where, $apply_limits = true, $rows = 3
 	/* status sql where */
 	if (get_request_var('status') != '-1') {
-		$sql_where .= ($sql_where != '' ? ' AND ':'WHERE ') . ' grid_load.status=?';
+		$sql_where .= ($sql_where != '' ? ' AND ':'WHERE ') . ' grid_load.status = ?';
```
Got an error in cacti.log

```
2025-05-13 16:04:42 - CMDPHP SQL Backtrace: (/plugins/grid/grid_lsload.php[45]:grid_view_load(), /plugins/grid/grid_lsload.php[569]:db_fetch_cell_prepared(), /lib/database.php[594]:db_execute_prepared())

2025-05-13 16:04:42 - CMDPHP ERROR: A DB Cell Failed!, Error: Unknown column 'grid_load.status' in 'where clause'
```
May need to change all `grid_load` to `gl` in query.

TheWitness replied:
This is fixed.  Thanks!

cwei-ca commented on cacti/plugins/grid/grid_bhosts.php:
```diff
@@ -1047,23 +1059,43 @@ function clearFilter() {
+			AND gh.host = gqh.host';
+	}
+
+	$total_rows = db_fetch_cell("SELECT COUNT($distinct gh.host)
```
Got error in cacti.log 

```
2025-05-13 16:02:50 - CMDPHP SQL Backtrace: (/plugins/grid/grid_bhosts.php[64]:grid_view_bhosts(), /plugins/grid/grid_bhosts.php[1083]:db_fetch_cell(), /lib/database.php[572]:db_fetch_cell_prepared(), /lib/database.php[594]:db_execute_prepared())

2025-05-13 16:02:50 - CMDPHP ERROR: A DB Cell Failed!, Error: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '?' at line 1
```

TheWitness replied:
This is fixed now.

TheWitness committed:
QA: Fixing two prepared statements causing errors

Signed-off-by: TheWitness <thewitness@cacti.net>

TheWitness committed:
QA: update escape() to encodeURIComponent() plus light PSR, JS memory leaks

Signed-off-by: TheWitness <thewitness@cacti.net>

TheWitness committed:
QA: Fixing some table aliases for grid_load and grid_hostinfo

Signed-off-by: TheWitness <thewitness@cacti.net>

TheWitness commented:
Okay @cwei-ca, Everything looking better now.  I corrected one additional escape() call in grid_bqueues.php.  There were also a few memory leaks that I fixed and some PSR too.  Appreciate your detailed review.

TheWitness commented:
This pull includes a fix that will be a part of Cacti 1.2.31.
https://github.com/Cacti/cacti/issues/6251

TheWitness commented:
Upstream patch is here.
https://github.com/Cacti/plugin_thold/commit/d913342df1b1c745b15b68f37cf65c815fbb2eda

TheWitness commented:
> Plz remove the change in html.php that was already in PR [50](https://github.com/IBM/ibm-spectrum-lsf-rtm-server/pull/50).

I can not do that as I was forked off of my local commit.  When you merge it, it won't break anything.
